src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -147,29 +151,31 @@ export class HuiDialogEditCard extends LitElement {
 
   private async _loadConfigElement(): Promise<void> {
     const conf = yaml.safeLoad(this._currentConfigYaml);
+
     const tag = conf.type.startsWith(CUSTOM_TYPE_PREFIX)
       ? conf.type.substr(CUSTOM_TYPE_PREFIX.length)
       : `hui-${conf.type}-card`;
-    const elClass = customElements.get(tag);
 
-    let elementConfig;
+    const elClass = customElements.get(tag);
+    let configElement;
     try {
-      elementConfig = await elClass.getConfigElement();
-      elementConfig.setConfig(conf);
-      elementConfig.hass = this.hass;
-      elementConfig.addEventListener("config-changed", (ev) =>
+      configElement = await elClass.getConfigElement();
+      configElement.setConfig(conf);
+      configElement.hass = this.hass;
+      configElement.addEventListener("config-changed", (ev) =>
         this._handleConfigChanged(ev.detail.config)
       );
-      this._elementConfig = elementConfig;
+      this._configElement = configElement;
     } catch (err) {
-      this._elementConfig = null;
+      this._configElement = null;
     }
 
     // This will center the dialog with the updated config Element
     fireEvent(this._dialog, "iron-resize");
   }
 
   private async _updateConfig(): Promise<void> {
+    this._currentConfigYaml = yaml.safeDump(this._previewEl.config);
```

Comment:
It seems weird that we use our preview element as the source of truth of what the current config is.